scripts/base/protocols/dns/main.zeek

```diff
@@ -428,6 +443,8 @@ event dns_request(c: connection, msg: dns_msg, query: string, qtype: count, qcla
 	c$dns$qtype       = qtype;
 	c$dns$qtype_name  = query_types[qtype];
 	c$dns$Z           = msg$Z;
+	c$dns$AD          = msg$AD;
```

Comment:
I'm unsure if we should set `AD` when processing `dns_request`: In requests it should be 0 IIUC.

We could instead consider to flag AD being set in requests (or CD being set in responses) as a weird, but can do as a follow up.

Reply:
RFC 6840 states that the AD bit can be set by the request or response
https://www.rfc-editor.org/rfc/rfc6840#section-5.7

Reply:
More info on why - The value set here will be overriden by the response value if both the request and reply are seen by Zeek. Definitely worth adding weird logs in the future.

Reply:
Thanks for re-linking that RFC.

> request and reply are seen by Zeek

Which should be the common case. What's there now I think is okay. As you mention, the `AD` column in the log will be overwritten by the response's AD flag and that's semi great.

I'm paging @ckreibich if we should solve this somehow differently in a follow-up (new column, or let script writers extend the log themselves), but I'll look into merging your this changeset as is.